### PR TITLE
feat: memory limiter unification write path

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

~~This PR consolidates three overlapping memory limiting mechanisms into a single unified implementation using `MemoryManager`.~~

~~**Background:** 
GreptimeDB previously had three memory limiters with overlapping responsibilities: 
(1) `max_in_flight_write_bytes` in Frontend (recursive size calculation, expensive), (2) `RequestMemoryLimiter` in HTTP/gRPC servers (atomic counter), and (3) `MemoryManager` for compaction. 
The Frontend limiter performed costly per-request data structure traversals while Server layer already protected network ingress.~~

~~**Changes:** 
Removed Frontend's `max_in_flight_write_bytes` option and `limiter.rs` , 
replaced Server's custom `RequestMemoryLimiter` with `MemoryManager` using 1KB granularity (`PermitGranularity::Kilobyte`) for fine-grained fairness in high-concurrency scenarios, 
and added comprehensive integration tests covering boundary conditions to verify alignment behavior .~~

**Impact:** 
Single unified memory control mechanism across all write paths, eliminated redundant per-request size calculations, consistent configuration model (`max_total_body_memory` for HTTP, `max_total_message_memory` for gRPC), and improved test coverage with accurate granularity alignment documentation.~~

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

File: tests-integration/tests/http.rs [hunk 1]
```diff
@@ -7114,3 +7117,139 @@ fn compress_vec_with_gzip(data: Vec<u8>) -> Vec<u8> {
...
+        .body(format!("sql={}", medium_insert))
+        .send()
+        .await;
+    assert_eq!(
+        res.status(),
+        StatusCode::OK,
+        "Medium request (~700 bytes) should succeed within aligned 1KB limit"
+    );
+
+    // Test 2: Large write request should be rejected (exceeds 500 bytes)
```
[the remark this answers is not in the document]
Comment (fengjiachun) [reply tests-integration/tests/http.rs]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/GreptimeTeam/greptimedb/pull/7437#discussion_r2630639651)

**Comment by sunng87** [comment]
Sounds good to me. And why we use separated memory limiters for gRPC and HTTP? Can we share the same permit pool for all interfaces? It can be easier to describe and config.

**Comment by fengjiachun** [comment]
> Sounds good to me. And why we use separated memory limiters for gRPC and HTTP? Can we share the same permit pool for all interfaces? It can be easier to describe and config.

Sounds good, I am working on this

**Comment by fengjiachun** [comment]
> Sounds good to me. And why we use separated memory limiters for gRPC and HTTP? Can we share the same permit pool for all interfaces? It can be easier to describe and config.

@sunng87 Already followed your advice. PTAL

**Comment by fengjiachun** [comment]
@MichaelScofield PTAL

**Comment by sunng87** [comment]
LGTM. And this is no longer a breaking issue with the update. We can consider it as an enhancement of previous max in-flight bytes limiter.
